src/ghazal_azadi_ex/ex04/walker.py

```diff
@@ -0,0 +1,54 @@
+__author__ = 'Ghazal_Azadi'
+__email__ = '@ghazal.azadi@nmbu.no'
+
+import random
+
+
+class Walker:
+
+    def __init__(self, position, home_position):
+        self.position = position
+        self.home_position = home_position
+        self.state = True
+        self.steps = 0
+
+    def is_at_home(self):
+
+        if self.position == self.home_position:
+            self.state = True
+        else:
+            self.state = False
+
+        return self.state
+
+    def move(self):
+        if not self.is_at_home():
+            rand = random.randint(0, 1)
+            if rand == 0:
+                self.position -= 1
+            else:
+                self.position += 1
+
+            self.steps += 1
+
+    def get_position(self):
+
+        return self.position
+
+    def get_steps(self):
+        return self.steps
+
+    def walking_process(self):
+        while not self.is_at_home():
+            Walker.move(self)
+        return Walker.get_steps(self)
```

Comment:
This is wrong.
Instead, create the walking_process function outside this class (as an independent function)

Reply:
why should not this function be part of the class?